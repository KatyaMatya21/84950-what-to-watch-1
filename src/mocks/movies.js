export default [
  {
    'id': 1,
    'name': `Fantastic Beasts: The Crimes of Grindelwald`,
    'previewImage': `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    'link': `movie-page.html`,
    'previewVideoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'genre': `Comedies`
  },
  {
    'id': 2,
    'name': `Bohemian Rhapsody`,
    'previewImage': `img/bohemian-rhapsody.jpg`,
    'link': `movie-page.html`,
    'previewVideoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'genre': `Crime`
  },
  {
    'id': 3,
    'name': `Macbeth`,
    'previewImage': `img/macbeth.jpg`,
    'link': `movie-page.html`,
    'previewVideoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'genre': `Documentary`
  },
  {
    'id': 4,
    'name': `We need to talk about Kevin`,
    'previewImage': `img/we-need-to-talk-about-kevin.jpg`,
    'link': `movie-page.html`,
    'previewVideoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'genre': `Dramas`
  },
  {
    'id': 5,
    'name': `What We Do in the Shadows`,
    'previewImage': `img/what-we-do-in-the-shadows.jpg`,
    'link': `movie-page.html`,
    'previewVideoLink': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'genre': `Horror`
  }
];
