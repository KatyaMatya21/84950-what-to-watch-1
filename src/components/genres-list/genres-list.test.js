import React from 'react';
import renderer from 'react-test-renderer';

import movies from '../../mocks/movies';

import GenresList from '../genres-list/genres-list.jsx';

describe(`GenresList component`, () => {

  it(`Correctly renders`, () => {
    const GenresListComponent = renderer
      .create(
          <GenresList
            movies={movies}
            activeItem={false}
            onClick={() => jest.fn()}
            selectGenre={() => jest.fn()}
          />,
          {
            createNodeMock: (element) => {
              if (element.type === `video`) {
                return {'current': {}};
              }
              return null;
            }
          }
      )
      .toJSON();
    expect(GenresListComponent).toMatchSnapshot();
  });

});
