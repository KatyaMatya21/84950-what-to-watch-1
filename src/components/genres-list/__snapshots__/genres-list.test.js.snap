// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenresList component Correctly renders 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <button
      className="catalog__genres-link"
      onClick={[Function]}
      type="button"
    >
      All genres
    </button>
  </li>
  <li
    className="catalog__genres-item "
  >
    <button
      className="catalog__genres-link"
      onClick={[Function]}
      type="button"
    >
      Comedies
    </button>
  </li>
  <li
    className="catalog__genres-item "
  >
    <button
      className="catalog__genres-link"
      onClick={[Function]}
      type="button"
    >
      Crime
    </button>
  </li>
  <li
    className="catalog__genres-item "
  >
    <button
      className="catalog__genres-link"
      onClick={[Function]}
      type="button"
    >
      Documentary
    </button>
  </li>
  <li
    className="catalog__genres-item "
  >
    <button
      className="catalog__genres-link"
      onClick={[Function]}
      type="button"
    >
      Dramas
    </button>
  </li>
  <li
    className="catalog__genres-item "
  >
    <button
      className="catalog__genres-link"
      onClick={[Function]}
      type="button"
    >
      Horror
    </button>
  </li>
</ul>
`;
